crates/cdk/src/cdk_database/mod.rs

```diff
@@ -124,8 +124,18 @@ pub trait WalletDatabase: Debug {
     /// Remove [`Keys`] from storage
     async fn remove_keys(&self, id: &Id) -> Result<(), Self::Err>;
 
-    /// Add [`Proofs`] to storage
-    async fn add_proofs(&self, proof_info: Vec<ProofInfo>) -> Result<(), Self::Err>;
+    /// Update the proofs in storage by adding new proofs or removing proofs by their Y value.
+    async fn update_proofs(
+        &self,
+        added: Vec<ProofInfo>,
+        removed_ys: Vec<PublicKey>,
+    ) -> Result<(), Self::Err>;
+    /// Set proofs as pending in storage. Proofs are identified by their Y value.
+    async fn set_pending_proofs(&self, ys: Vec<PublicKey>) -> Result<(), Self::Err>;
+    /// Reserve proofs in storage. Proofs are identified by their Y value.
+    async fn reserve_proofs(&self, ys: Vec<PublicKey>) -> Result<(), Self::Err>;
+    /// Unreserve proofs in storage. Proofs are identified by their Y value.
+    async fn unreserve_proofs(&self, ys: Vec<PublicKey>) -> Result<(), Self::Err>;
     /// Get proofs from storage
```

Comment:
Proofs can got from state `UNSPENT`` -> `PENDING` as well as the reverse the same is true for the `RESERVED` state `UNSPENT` -> `RESERVED` and back. The `unreserve_proof` fn here can bring a proof from either `RESERVED` or `PENDING` to `UNSPENT` so i suggest we just change the name to reflect that.

Reply:
Do you have a suggested name? `set_unspent_proofs`?